test/controllers/underpayments/RemoveUnderpaymentDetailsControllerSpec.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.underpayments
+
+import base.ControllerSpecBase
+import controllers.actions.FakeDataRetrievalAction
+import forms.underpayments.RemoveUnderpaymentDetailsFormProvider
+import mocks.repositories.MockSessionRepository
+import models.UserAnswers
+import models.underpayments.UnderpaymentDetail
+import pages.underpayments.UnderpaymentDetailSummaryPage
+import play.api.http.Status
+import play.api.mvc.{AnyContentAsFormUrlEncoded, Result}
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{charset, contentType, defaultAwaitTimeout, redirectLocation, status}
+import views.html.underpayments.RemoveUnderpaymentDetailsView
+
+import scala.concurrent.Future
+
+
+class RemoveUnderpaymentDetailsControllerSpec extends ControllerSpecBase {
+
+
+  trait Test extends MockSessionRepository {
+    private lazy val removeUnderpaymentDetailsView: RemoveUnderpaymentDetailsView = app.injector.instanceOf[RemoveUnderpaymentDetailsView]
+
+    val userAnswers: Option[UserAnswers] = Some(UserAnswers("credId"))
+    private lazy val dataRetrievalAction = new FakeDataRetrievalAction(userAnswers)
+
+    val underpaymentType = "B00"
```

Comment:
this needs to be a sequence of all the underpayment types, and you should check it's fine for all of them